### PR TITLE
Ensure that the result of trivial phi simplification is alive     

### PR DESCRIPTION
**Pull request by ptomin**
- Create `SsaTransform` test with loop with break and return. It failed to simplify trivial phi functions
- Ensure that the result of trivial phi simplification is alive
This result could be removed during recursive trivial phi simplification
Simplify phi function until identifier is not in the list of
removed phi identifiers

### CODE REVIEW

File: src/Decompiler/Analysis/SsaTransform.cs [hunk 1]
```diff
@@ -1357,29 +1357,19 @@ private static void GeneratePhiFunction(SsaIdentifier phi, SsaIdentifier[] sids)
...
-                        continue;
-                    if (same != null)
-                    {
-                        sid = null;
-                        return false;
-                    }
-                    same = op;
-                }
-                if (same == null)
+                if (!phiFunc.Arguments.Where(a => a != phi.Identifier).Any())
```
Comment (uxmal) [inline src/Decompiler/Analysis/SsaTransform.cs]:
```suggestion
                if (!phiFunc.Arguments.Any(a => a != phi.Identifier))
```
may be slightly more performant.

File: src/Decompiler/Analysis/SsaTransform.cs [hunk 2]
```diff
@@ -1357,29 +1357,19 @@ private static void GeneratePhiFunction(SsaIdentifier phi, SsaIdentifier[] sids)
...
-                        continue;
-                    if (same != null)
-                    {
-                        sid = null;
-                        return false;
-                    }
-                    same = op;
-                }
-                if (same == null)
+                if (!phiFunc.Arguments.Any(a => a != phi.Identifier))
```
Comment (uxmal) [inline src/Decompiler/Analysis/SsaTransform.cs]: The double negative is confusing here, now. Isn't this the same as saying 
```suggestion
                if (phiFunc.Arguments.All(a => a == phi.Identifier))
```

Comment (ptomin) [reply src/Decompiler/Analysis/SsaTransform.cs]: Changed. See 9f76399

**Comment by ptomin** [comment]
@uxmal I have done requested changes

**Pull request merged**